### PR TITLE
Enforce no mutating functions in protocols

### PR DESCRIPTION
**Pull request by dvosscricut**
It's the law!

### CODE REVIEW

File: Sources/FishyJoesCore/TranslatedTypes/TranslatedProtocol.swift [hunk 1]
```diff
@@ -58,36 +58,37 @@ struct TranslatedProtocol: TranslatedType {
...
     func enforceProtocolThrows() {
         let nonThrowingMethods = methods.filter { !$0.isThrowing }
         guard nonThrowingMethods.isEmpty else {
             var nameSpace = ""
-            if let ns = nonThrowingMethods.first!.definedIn?.name {
+            if let ns = nonThrowingMethods.first?.definedIn?.name {
                 nameSpace = "\(ns)."
             }
-            fatalError("☠️ Error on \(nameSpace)\(nonThrowingMethods.first!.name): All Protocol methods exported through FishyJoes must be throwing, it's the law 👮!")
+            fatalError("☠️ Error on \(nameSpace)\(nonThrowingMethods.first?.name ?? "?"): All Protocol methods exported through FishyJoes must be throwing, it's the law 👮!")
```
Comment (cobbal) [inline Sources/FishyJoesCore/TranslatedTypes/TranslatedProtocol.swift:71]: nit: This is what an `if let` is for.

**Review by cobbal — APPROVED** [review]

File: Sources/FishyJoesCore/TranslatedTypes/TranslatedProtocol.swift (hunk 1 above)
Comment (dvosscricut) [reply Sources/FishyJoesCore/TranslatedTypes/TranslatedProtocol.swift:71]: I prefer the terser syntax of ?? "?"

Comment (cobbal) [reply Sources/FishyJoesCore/TranslatedTypes/TranslatedProtocol.swift:71]: It's not terser, and it has a never executed code path

Comment (cobbal) [reply Sources/FishyJoesCore/TranslatedTypes/TranslatedProtocol.swift:71]: To be clear, I'm saying replace the guard with an if let that binds the thing you care about

**Pull request merged**